src/multiprocess/multiprocess_memory_limit.c

```
@@ -787,6 +787,17 @@ void lock_shrreg() {
                 }
                 continue;  // slow wait path
             }
+            // RECOVERY: unlock_shrreg() race where owner_pid was cleared
+            // but sem_post() never executed 
+            // if owner_pid == 0 and we are still timing out, the semaphore
+            // is stuck locked with no owner. Force a post to unstick it.
+            if (current_owner == 0 && trials > 5) {
+                LOG_WARN("Detected stuck semaphore with owner_pid == 0 "
```

Cppcheck (2.21.0)

[warning] If resource allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfResources)